build.sbt

```diff
@@ -552,7 +552,7 @@ lazy val `testkit-scaladsl` = (project in file("testkit/scaladsl"))
     `persistence-core`               % "compile;test->test",
     `persistence-scaladsl`           % "compile;test->test",
     `persistence-cassandra-scaladsl` % "compile->test;test->test",
-    `persistence-jdbc-scaladsl`      % Test
+    `persistence-jdbc-scaladsl`      % "compile;test->test"
```

Comment:
I just realised that this looks wrong: If I am a Cassandra user I shouldn't get `persistence-jdbc-scaladsl` as a transitive dependency. 
Maybe it should be `provided`? But then there'd be a runtime error for cassandra users invoking `ServiceTest` since `import com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents` will raise a `ClassNotFound`.

Reply:
We discussed this approach offline and it sounded OK to me. But I'm not sure. 

I guess it's the least bad of the options.

Reply:
Yes. I saw only 3 variants:
1)
Eagerly initialize `slickProvider` in `com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents`
=> **lazy property with unlazy behavior** :disappointed:
Was discussed in https://github.com/lagom/lagom/pull/2887
2)
Add eager initialization in `com.lightbend.lagom.scaladsl.testkit.ServiceTest#startServer`
=> **`testkit-scaladsl` depends on `persistence-jdbc-scaladsl` for compile scope** :disappointed:
3)
Try to implement a second approach by reflection
=> **forget about beautiful source code** :joy:

Reply:
I think I'm beginning to lean towards `Eagerly initialize slickProvider in com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents
=> lazy property with unlazy behavior 😞`